Add services.yaml entry for new WeMo Humidifier platform service

sqldiablo opened the pull request:
## Description:
Add an entry to services.yaml describing the wemo_set_humidity service offered by the new WeMo Humidifier platform

**Related issue (if applicable):** addition to #17996

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**

sqldiablo committed:
Fix Vera climate component to use correct states

Changed the Vera climate component so it uses the STATE_* states from the base climate component. This will allow it to work with Google Assistant.

sqldiablo committed:
Merge pull request #1 from home-assistant/dev

Catch-up PR

sqldiablo committed:
Wemo Humidifier - Initial Commit

sqldiablo committed:
WeMo Humidifier - First draft of component

sqldiablo committed:
WeMo Humidifier - Removed direct IO from property

sqldiablo committed:
WeMo Humidifier - Trivial comment change

sqldiablo committed:
Added myself as codeowner for WeMo

sqldiablo committed:
WeMo Humidifier - Fixed various syntax & lint issue

sqldiablo committed:
WeMo Humidifier - Small comment addition

sqldiablo committed:
WeMo Humidifier - Fix TypeError: 'WemoHumidifier' object is not iterable

sqldiablo committed:
WeMo Humidifier - Rename set humidity service

sqldiablo committed:
WeMo Humidifier - Add to .coveragerc

sqldiablo committed:
WeMo Humidifier - Fixed lint/pylint issues

sqldiablo committed:
WeMo Humidifier - First round of requested changes

sqldiablo committed:
WeMo Humidifier - Round two of requested changes

sqldiablo committed:
WeMo Humidifier - Third round of requested changes

sqldiablo committed:
WeMo Humidifier - Fixed whitespace issue on dict comprehension

sqldiablo committed:
WeMo Humidifier - Fourth round of requested changes

sqldiablo committed:
WeMo Humidifier - Corrected typo in async_add_executor_job call

sqldiablo committed:
WeMo Humidifier - Fixed spacing before inline comments

sqldiablo committed:
WeMo Humidifier - Added entry to services.yaml

MartinHjelmare commented:
Please rebase your branch on dev to only add the last commit.

MartinHjelmare commented:
Take for habit to always work in a separate feature branch, ie not `dev`. That will make it a lot easier to continue working on new things.

sqldiablo committed:
Catch-up PR (#2)

* Bumped version to 0.81.0b0

* Update frontend to 20181023.0

* Update translations

* Geo location trigger added (#16967)

* zone trigger supports entity id pattern

* fixed lint error

* fixed test code

* initial version of new geo_location trigger

* revert to original

* simplified code and added tests

* refactored geo_location trigger to be based on a source defined by the entity

* amended test cases

* small refactorings

* Add lovelace websocket get and set card (#17600)

* Add ws get, set card

* lint+fix test

* Add test for set

* Added more tests, catch unsupported yaml constructors

Like !include will now give an error in the frontend.

* lint

* Fixed issue #16903 re exception with multiple simultanious writes (#17636)

Reworked tests/components/emulated_hue/test_init.py to not be
dependent on the specific internal implementation of util/jsonn.py

* Update flux library version (#17677)

* Update limitlessled to 1.1.3 (#17703)

* Bug fix for clicksend (#17713)

* Bug fix

Current version causes 500 error since it is sending an array of from numbers to ClickSend. Changing the from number to 'hass' identifies all messages as coming from Home Assistant making them more recognisable and removes the bug.

* Amendment

Changed it to use 'hass' as the default instead of defaulting to the recipient which is the array. Would have worked if users set their own name but users who were using the default were experiencing the issue.

* Added DEFAULT_SENDER variable

* Bumped version to 0.81.0b1

* Update frontend to 20181024.0

* Elk-M1 climate (#17679)

* Initial climate for Elk-M1.

* Tidy

* fix hound error

* fix hound error

* Lovelace ws: add card (#17730)

* Change set to update

* Add 'add card'

* Woof.

* Add cover to supported platforms (#17725)

* Remove day (fixes #17741) (#17743)

* Bumped version to 0.81.0b2

* Bump frontend to 20181026.0

* Update translations

* Fixed an incorrect reference in the entity registry (#17775)

* Update HAP-python to 2.3.0 (#17778)

* Update HAP-python to 2.3.0

* Fix tests

* SMHI weather component not showing correct values in current forecast (#17783)

* fixes not showing current forecast correctly

* Update config_flow.py

* Update smhi.py

* Update requirements_all.txt

* Update requirements_test_all.txt

* Fix unloading an entry can leave states around (#17786)

* Add test that tests unloading on remove

* Add more test things

* Untangle entity remove code from entity platform

* Don't add default implementation of async_will_remove

* Keep entity weakref alive

* Bumped version to 0.81.0

* frontend bump

* Switch to using Client from twilio.rest rather than the deleted TwilioRestClient (#17885)

* bump frontend to 20181026.1

* Fixes an issue with OpenUV config import failing (#17831)

* Fixes an issue with OpenUV config import failing

* Update

* Update __init__.py

* Update config_flow.py

* Allow a list ofr update entity (#17860)

* Allow a list ofr update entity

* Update services.yaml

* Update services.yaml

* Move migrate to separate WS command (#17890)

* Bumped version to 0.81.1

* Axis - prepare for config entry (#17566)

Make component more in line with other more modern components in preparation for config entry support.

* Upgrade pytest to 3.9.3 (#17921)

* Extract entity ids from all templates (#17902)

* Make light.yeelight stop doing IO when accessing properties (#17917)

* Fixes for upnp-component/#17753 and missing hass-data when only setup from config entry (#17868)

* Upgrade to async_upnp_client==0.13.0, fixing #17753

* Fix missing 'local_ip' when upnp-component itself is not setup, but ConfigEntry is

* Fix for verify_ssl in the pi_hole sensor. (#17910)

* Permissions improv (#17811)

* Break up permissions file.

* Granular entity permissions

* Add "all" entity permission

* Lint

* Fix types

* Fix incorrect chevy discovery (#17942)

* Fix venv check (#17939)

* Fix venv check

* Lint

* Disable upnp from being discovered (#17937)

* Fix controller not being stored when setup fails and sequentially fails the retry functionality (#17927)

* Pass hass_config to load_platform (#17952)

* Pass hass_config to load_platform

* Fix tests

* Lint

* Fix controller not being stored when setup fails and sequentially fails the retry functionality (#17927)

* Pass hass_config to load_platform (#17952)

* Pass hass_config to load_platform

* Fix tests

* Lint

* Bumped version to 0.81.2

* Added codeowner for bt_smarthub (#17947)

* Fixed copy-paste errors (#17948)

* Fix spelling error in log output (#17963)

* Fix operation mode for Alexa thermostat (#17972)

* Change source, add attributes, and improve state of DirecTV (#17536)

* Enhancements for DirecTV media player

Following enhancements have been made:

1. Added debug logging
2. Added ability to change channel using select_source service of the remote platform.
3. State will now show paused if a recorded program is paused, for live TV playing will always be returned.
4. Added the following attributes:
    a. media_position: current position of the media (in seconds)
    b. media_position_updated_at: timestamp when media_position was updated.
   c. source: current source (channel).
   d. media_isbeingrecorded: if current media is being recorded or not.
   e. media_rating: TV/Movie rating of the media
   f. media_recorded: if current media is recorded or live TV
   g. media_starttime: Timestamp media was aired

Reordered properties to follow same order as how they are in __init__.py of remote platform.

* Fixed error and cleaned up few items

Fixed an issue when determining if a program is recorded or not.
Cleaned up some coding.

* Attribute last position update only updated when position changed.

The attribute media_position_updated_at will only be updated if the position changed (thus media is playing for recorded or live TV).
Added assumed_state; will be set to False if in standby or when a recorded show is watched. For live TV it will be set to True.

* Added some empty lines for easier reading

Added some empty lines before returns to improve readability.

* Seperated words in constants

Seperated the words in constants.

* Fix _lastupdate to _last_update

Split words in _lastupdate to _last_update as I missed it.

* Update requests to 2.20.0 (#17978)

* Don't use keyset (#17984)

* Update requests to 2.20.0 (#17978)

* Don't use keyset (#17984)

* Update Alexa tests to async syntax (#17965)

See https://github.com/home-assistant/home-assistant/issues/12614

* Météo-france sensor : current weather and 1 hour rain forecast (#17773)

* 🌧 Meteo France rain forecast

* do not fail on wrong config

* Update name of sensor class

* do not show sensor if not working

* Update .coveragerc

* ability to submit insee location code without final 0 needed by meteo-france

* Lynting

* more lynting

* update comment

* block comment linting

* reducing length of long lines

* linting

* Update météo-france platform

Now work with pypi package and monitored conditions

* remove error log

* Update requirements_all.txt

* Increase scan interval to 5 minutes

* Update meteo_france according to review

* better error handling of location missing some monitored conditions

* fix lint error

* moving error log

Errors are now catched at sensor initialization and state is set when updating the data

* Update updating of sensor

* Updated ring_doorbell dependency to 0.2.2 (#17945)

- Resolves an issue with the sensor platform for Ring Spotlight Cam devices

* Update Avi-On to work with latest API (#17780)

* Update Avi-On to work with the API.

* Use voluptuous defaults instead of .get()

* Bump library version.

* Remove unnecessary voluptuous defaults. Fix manually-configured devices.

API-discovered devices are already Avion objects, but manually-configured devices need to be instantiated as Avion objects first.

* Use .get() where appropriate.

* Remove default

* Mill service (#17971)

* Mill service

* style

* Refactor Alexa API, fix thermostats (#17969)

* Refactor Alexa API to use objects for requests

This introduces _AlexaDirective to stand in for the previous model of passing
basic dict and list data structures to and from handlers. This gives a more
expressive platform for functionality common to most or all handlers.

I had two use cases in mind:

1) Most responses should include current properties. In the case of locks and
thermostats, the response must include the properties or Alexa will give the
user a vague error like "Hmm, $device is not responding." Locks currently work,
but thermostats do not. I wanted a way to automatically include properties in
all responses. This is implemented in a subsequent commit.

2) The previous model had a 1:1 mapping between Alexa endpoints and Home
Assistant entities. This works most of the time, but sometimes it's not so
great. For example, my Z-wave thermostat shows as three devices in Alexa: one
for the temperature sensor, one for the heat, and one for the AC. I'd like to
merge these into one device from Alexa's perspective. I believe this will be
facilitated with the `endpoint` attribute on `_AlexaDirective`.

* Include properties in all Alexa responses

The added _AlexaResponse class provides a richer vocabulary for handlers.

Among that vocabulary is .merge_context_properties(), which is invoked
automatically for any request directed at an endpoint. This adds all supported
properties to the response as recommended by the Alexa API docs, and in some
cases (locks, thermostats at least) the user will get an error "Hmm, $device is
not responding" if properties are not provided in the response.

* Fix setting temperature with Alexa thermostats

Fixes https://github.com/home-assistant/home-assistant/issues/16577

* Upgrade aiofreepybox (#17989)

- Add features to get connection status and stats
 - Improve error reporting
 - Fix issue with the authentication file (#16934)

* fix naming bug (#17976)

* Update frontend to 20181030.0

* Update translations

* Reverse out change #14234 BOM Weather throttle fix (#17468)

* Reverse out change #14234 BOM Weather throttle fix

Reverted back to original throttle code to ensure sensors are updated on time.

* Fixed lint issues

* Review as a commit

* Use last_updated for attributes

* lint

* lint

* RFC: Static templates match no entities, not all (#17991)

* Static templates match no entities, not all

* Clean up test values

* Switch mailgun webhooks to the new Mailgun webhook api (#17919)

* Switch mailgun webhooks to the webhook api

* Change mailgun strings to indicate application/json is in use

* Lint

* Revert Changes to .translations.

* Don't fail if the API key isn't set

* Change hebrew date at sunset (#17449)

* Change date at sunset

* Fix tests to actually run and add fix to component

* Make tests pass

* Use get_astral_event_next instead of get_astral_event_date

* Revert to using get_astral_event_date

* Make tox happy: reset state on tearDown

* Import homeassistant domain instead of hardcoding it (#17985)

* Allow jinja namespace command to work. (#18011)

* Support for WeMo Humidifier (#17996)

* Fix Vera climate component to use correct states

Changed the Vera climate component so it uses the STATE_* states from the base climate component. This will allow it to work with Google Assistant.

* Wemo Humidifier - Initial Commit

* WeMo Humidifier - First draft of component

* WeMo Humidifier - Removed direct IO from property

* WeMo Humidifier - Trivial comment change

* Added myself as codeowner for WeMo

* WeMo Humidifier - Fixed various syntax & lint issue

* WeMo Humidifier - Small comment addition

* WeMo Humidifier - Fix TypeError: 'WemoHumidifier' object is not iterable

* WeMo Humidifier - Rename set humidity service

* WeMo Humidifier - Add to .coveragerc

* WeMo Humidifier - Fixed lint/pylint issues

* WeMo Humidifier - First round of requested changes

* WeMo Humidifier - Round two of requested changes

* WeMo Humidifier - Third round of requested changes

* WeMo Humidifier - Fixed whitespace issue on dict comprehension

* WeMo Humidifier - Fourth round of requested changes

* WeMo Humidifier - Corrected typo in async_add_executor_job call

* WeMo Humidifier - Fixed spacing before inline comments

* Fix Bloomsky api call (#18016)

* Upgrade pyipma (#17992)

* bump dependency version

* Add more context to debug message

Co-Authored-By: dgomes <diogogomes@gmail.com>

* shorten debug messages

* Async version of melissa (#17721)

* rebase upstream

* Fixed tests

* Fixing lint

MartinHjelmare commented:
There are still a lot of commits.

sqldiablo commented:
Sorry, I'm having a heck of a time getting that to work right...

sqldiablo committed:
Revert "Catch-up PR (#2)"

This reverts commit 864e91efcd08bf4e6a3807e75a50ecbcd0569fcd.
